core/src/main/java/com/linecorp/armeria/common/multipart/NoopSubscription.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.multipart;
+
+import org.reactivestreams.Subscription;
+
+final class NoopSubscription implements Subscription {
```

Comment:
We have the same class in the stream package. Should we move this into the internal package and use both places? 🤔

Reply:
Also, we can make this as an enum singleton.

Reply:
> We have the same class in the stream package. Should we move this into the internal package and use both places? 🤔

`NoopSubscription` doesn't have a special logic. So I thought it is safe to copy. And `NoopSubscription` is ephemeral. `Mulit` and related classes will be removed later once Armeria adopts other Reactive Streams library.

> Also, we can make this as an enum singleton.

Yeap.